### PR TITLE
Add AMD python package ROCm5.1.1+torch1.11

### PR DESCRIPTION
**Pull request by PeixuanZuo**
**Description**: Describe your changes.

Add AMD python package ROCm5.11 + torch1.11
Delete AMD python package ROCm5.01+torch1.11
Simplify python3.6 related on AMD Dockerfile because we don't use python3.6 anymore.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Review by snnn — DISMISSED** [review]
LGTM

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-rocm.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ stages:
             --build-arg DEVTOOLSET_ROOTPATH=/opt/rh/devtoolset-10/root
             --build-arg PREPEND_PATH=/opt/rh/devtoolset-10/root/usr/bin:
             --build-arg LD_LIBRARY_PATH_ARG=/opt/rh/devtoolset-10/root/usr/lib64:/opt/rh/devtoolset-10/root/usr/lib:/opt/rh/devtoolset-10/root/usr/lib64/dyninst:/opt/rh/devtoolset-10/root/usr/lib/dyninst:/usr/local/lib64:/usr/local/lib
-          Repository: onnxruntimetrainingrocmbuild-torch1.10.0-rocm5.0.1
+          Repository: onnxruntimetrainingrocmbuild-torch1.11.0-rocm5.1.1
```
Comment (baijumeswani) [inline tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-rocm.yml:38]: I think we can remove the torch version (but keep the rocm version) in the repository name and delete all the old repositories that have the torch version string in the repository name.

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5_1_1 [hunk 2]
```diff
@@ -191,7 +184,7 @@ CMD ["/bin/bash"]
 #Build manylinux2014 docker image end
 
 ARG PYTHON_VERSION=3.7
-ARG TORCH_VERSION=1.11.0
+ARG TORCH_VERSION=1.10.0
```
Comment (baijumeswani) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5_1_1:187]: Why do we change default `TORCH_VERSION` to `1.10.0`? Shall we keep it at `1.11.0`?

**Review by baijumeswani — COMMENTED** [review]
Thanks for the pull request. I just have two very small comments. But otherwise the pull request looks good.

File: tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-rocm.yml (hunk 1 above)
Comment (PeixuanZuo) [reply tools/ci_build/github/azure-pipelines/orttraining-py-packaging-pipeline-rocm.yml:38]: If each rocm version corresponds to only one pytorch version in our python package pipeline, I think we can change it.

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5_1_1 (hunk 2 above)
Comment (PeixuanZuo) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5_1_1]: Yes, I think it should keep 1.11.0

**Review by baijumeswani — APPROVED** [review]

**Pull request merged**